src/script.js

```diff
@@ -22,31 +22,74 @@ export function config (filePath) {
 }
 
 export function load (runfilePath, config, logger, requirer, access) {
-  // try to load babel-register
-  try {
-    logger.log('Requiring babel-register...')
-    if (config['babel-register']) {
+  let babelEnabled = false
+  let tsEnabled = false
+  let hasJsRunfile = false
+  let hasTsRunfile = false
+  let importError = null
+
+  if (config['babel-register']) {
+    try {
       requirer(config['babel-register'])
-    } else {
+      babelEnabled = true
+    } catch (error) {
+      importError = error
+    }
+  } else if (config['ts-register']) {
+    try {
+      requirer(config['ts-register'])
+      tsEnabled = true
+    } catch (error) {
+      importError = error
+    }
+  } else {
+    try {
       requirer('./node_modules/babel-register')
+      babelEnabled = true
+    } catch (error) {
+      importError = error
     }
-  } catch (error) {
-    logger.log('Requiring failed. Fallback to pure node.')
-    if (config['babel-register']) {
-      throw error
+
+    if (!babelEnabled) {
```

Comment:
There is a lot of conditions to check which register should be loaded and when it should fallback to pure node. I'm wondering, maybe runjs shouldn't do automatic loading of `babel-register` or `ts-node/register` but it should strictly depend on `runjs` config from `package.json`. We could introduce in `package.json`:

```json
"runjs": {
  "transpiler": "babel-register"
}
```

and.. if `transpiler` is found it use given transpiler if not it fallbacks to pure node. In this way whole behaviour is more clear, reduces lines of code and it doesn't have to do checks if it can load some register and then fallback (which can be wasted time for someone who use pure node).

What do you think ?

So config can be also:
```json
"runjs": {
  "transpiler": "./node_modules/some_package/node_modules/babel-register"
}
```
or
```json
"runjs": {
  "transpiler": "ts-node/register",
  "runfile": "./runfile.ts"
}
```